pkg/log/log.go

```diff
@@ -149,6 +158,82 @@ func (l *OsqueryLogAdapter) logInfoAboutUnrecognizedProcessLockingPidfile(p []by
 	level.Debug(l.logger).Log(append(processInfo, "msg", "detected non-osqueryd process using pidfile")...)
 }
 
+// runAndLogPs runs ps filtering on the given PID, and logs the output.
+func (l *OsqueryLogAdapter) runAndLogPs(pid int) {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	pidStr := strconv.Itoa(pid)
+	cmd := exec.Command("ps", "-p", pidStr, "-o", "user,pid,ppid,pgid,stat,time,command")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		level.Debug(l.logger).Log(
+			"msg", "error running ps on non-osqueryd process using pidfile",
+			"pid", pid,
+			"err", err,
+		)
+		return
+	}
+
+	level.Debug(l.logger).Log(
+		"msg", "ran ps on non-osqueryd process using pidfile",
+		"pid", pid,
+		"output", string(out),
+	)
+}
+
+// runAndLogLsofByPID runs lsof filtering on the given PID, and logs the output.
+func (l *OsqueryLogAdapter) runAndLogLsofByPID(pid int) {
+	if runtime.GOOS == "windows" {
+		return
+	}
+
+	pidStr := strconv.Itoa(pid)
+	cmd := exec.Command("lsof", "-R", "-p", pidStr)
```

Comment:
Probably add `-n`

Reply:
```
       -n   inhibits the conversion of network numbers to host names for
            network files.  Inhibiting conversion may make lsof run faster.
            It is also useful when host name lookup is not working properly.
```

Happy to add it but would like to know why -- just for the performance impact, or another reason?

Reply:
General performance, yes. We don't need name resolution and the old skool unix habit is to always turn it off.